File: backend/app/schemas/prescription.py
from datetime import datetime
from pydantic import BaseModel

from app.models.prescription import PrescriptionStatus


class PrescriptionItemOut(BaseModel):
    id: int
    extracted_medication_name: str
    matched_medication_id: int | None
    dosage_text: str | None
    frequency_text: str | None
    duration_text: str | None
    confidence_score: float
    pharmacist_confirmed: bool

    class Config:
        from_attributes = True


class PrescriptionItemUpdate(BaseModel):
    matched_medication_id: int | None = None
    dosage_text: str | None = None
    frequency_text: str | None = None
    duration_text: str | None = None
    pharmacist_confirmed: bool | None = None


class PrescriptionOut(BaseModel):
    id: int
    customer_id: int
    pharmacist_id: int | None
    image_path: str
    raw_ocr_text: str | None
    status: PrescriptionStatus
    pharmacist_notes: str | None
    created_at: datetime
    reviewed_at: datetime | None
    items: list[PrescriptionItemOut] = []

    class Config:
        from_attributes = True


class PrescriptionReviewUpdate(BaseModel):
    status: PrescriptionStatus
    pharmacist_notes: str | None = None


class SafetyFlag(BaseModel):
    type: str           # duplicate_medication | drug_interaction | allergy | special_population
    severity: str        # info | warning | critical
    message_ar: str


class PrescriptionAnalysisResult(BaseModel):
    """Returned right after OCR + AI extraction, before pharmacist review."""
    prescription: PrescriptionOut
    safety_flags: list[SafetyFlag] = []
    disclaimer_ar: str = (
        "هذا تحليل أولي بالذكاء الاصطناعي لمساعدتك على الفهم فقط، وهو غير نهائي ولا يحل محل "
        "مراجعة الصيدلاني المرخّص لوصفتك الأصلية قبل صرف أي دواء."
    )

File: frontend/types/prescription.ts
export type PrescriptionStatus = "pending" | "analyzed" | "reviewed" | "dispensed" | "rejected";
export type SafetySeverity = "info" | "warning" | "critical";

export interface SafetyFlag {
  type: string;
  severity: SafetySeverity;
  message_ar: string;
}

export interface PrescriptionItem {
  id: number;
  extracted_medication_name: string;
  matched_medication_id: number | null;
  dosage_text: string | null;
  frequency_text: string | null;
  duration_text: string | null;
  confidence_score: number;
  pharmacist_confirmed: boolean;
}

export interface Prescription {
  id: number;
  customer_id: number;
  pharmacist_id: number | null;
  image_path: string;
  raw_ocr_text: string | null;
  status: PrescriptionStatus;
  pharmacist_notes: string | null;
  created_at: string;
  reviewed_at: string | null;
  items: PrescriptionItem[];
}

export interface PrescriptionAnalysisResult {
  prescription: Prescription;
  safety_flags: SafetyFlag[];
  disclaimer_ar: string;
}

export interface PrescriptionReviewPayload {
  status: PrescriptionStatus;
  pharmacist_notes?: string;
}

export interface PrescriptionItemUpdatePayload {
  matched_medication_id?: number | null;
  dosage_text?: string;
  frequency_text?: string;
  duration_text?: string;
  pharmacist_confirmed?: boolean;
}
